src/AmsToMqttBridge.ino

```diff
@@ -462,6 +465,154 @@ void readHanPort() {
 						String msg;
 						serializeJson(json, msg);
 						mqtt.publish(config.getMqttPublishTopic(), msg.c_str());
+					// 
+					// Start DOMOTICZ
+					//
+					} else if(config.getMqttPayloadFormat() == 2) {
+						StaticJsonDocument<512> json;
+						hanToJson(json, data, hw, temperature);
+						if (Debug.isActive(RemoteDebug::INFO)) {
+							debugI("Sending data to MQTT");
+							if (Debug.isActive(RemoteDebug::DEBUG)) {
+								serializeJsonPretty(json, Debug);
+							}
+						}
+						String msg;
+						serializeJson(json, msg);
+						mqtt.publish(config.getMqttPublishTopic(), msg.c_str());     // keep for now...
+						//
+						// Special MQTT messages for DOMOTIZ (https://www.domoticz.com/wiki/MQTT)
+						// -All messages should be published to topic "domoticz/in"
+						//
+						//  message msg_PE : send active power and and cumulative energy consuption to  virtual meter "Electricity (instant and counter)"
+						//
+						//      /json.htm?type=command&param=udevice&idx=IDX&nvalue=0&svalue=POWER;ENERGY
+						//
+						//       MQTT sample message:    {"command": "udevice",  "idx" : IDX , "nvalue" : 0, "svalue" : "POWER;ENERGY"}   
+						//         IDX = id of your device (This number can be found in the devices tab in the column "IDX")
+						//         POWER = current power (Watt)
+						//         ENERGY = cumulative energy in Watt-hours (Wh) This is an incrementing counter.
+						//               (if you choose as type "Energy read : Computed", this is just a "dummy" counter, not updatable because it's the result of DomoticZ calculs from POWER)
+						//
+						//  message msg_V1 : send Voltage of L1 to virtual Voltage meter
+						//
+						//	      /json.htm?type=command&param=udevice&idx=IDX&nvalue=0&svalue=VOLTAGE
+						//
+						//       MQTT sample message:    {"command": "udevice",  "idx" : IDX , "nvalue" : 0, "svalue" : "VOLTAGE"}   
+						//         IDX = id of your device (This number can be found in the devices tab in the column "IDX")
+						//         VOLTAGE = Voltage (V)
+						//  
+			
+						int idx1 = config.getDomoELIDX();
+						// TODO, this should be configurable....  
+						if (idx1 > 0) {
+							String PowerEnergy;
+							int p;
+							// double energy = config.getDomoEnergy();
+							double tmp_energy;
+							StaticJsonDocument<200> json_PE;
+							p = json["data"]["P"].as<int>();
```

Comment:
Get "P" with data.getActiveImportPower()

Reply:
Agree.

Reply:
implemented.  Similar changes  for voltage and current as well.